packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -230,11 +257,12 @@ export default class Proxy {
       vmNetworksTimeout
     )
     await timeout.call(
-      xapi._waitObjectState(vm.guest_metrics, guest_metrics =>
-        networkConfiguration === undefined
-          ? guest_metrics.networks['0/ip'] !== undefined
-          : guest_metrics.networks['0/ip'] === networkConfiguration.ip
-      ),
+      xapi._waitObjectState(vm.guest_metrics, guest_metrics => {
+        const ip = extractIp(guest_metrics.networks)
+        return networkConfiguration === undefined
+          ? ip !== undefined
+          : ip === networkConfiguration.ip
```

Comment:
Wouldn't it be better in this case to check that `guest_metrics.networks` contains the expected IP address?